test/e2e/specs/realrp/vnetpeering.go

```diff
@@ -0,0 +1,125 @@
+package realrp
+
+import (
+	"context"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-06-01/network"
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/resources"
+
+	"github.com/openshift/openshift-azure/pkg/api/2018-09-30-preview/api"
+	"github.com/openshift/openshift-azure/pkg/fakerp/client"
+	"github.com/openshift/openshift-azure/test/clients/azure"
+	tlog "github.com/openshift/openshift-azure/test/util/log"
+)
+
+var _ = Describe("Peer Vnet tests [Vnet][Real][LongRunning]", func() {
+	var (
+		ctx          = context.Background()
+		cli          *azure.Client
+		rg           = os.Getenv("RESOURCEGROUP")
+		vnetPeerName = "vnetPeer"
+		region       = os.Getenv("AZURE_REGION")
+		clientID     = os.Getenv("AZURE_CLIENT_ID")
+		clientSecret = os.Getenv("AZURE_CLIENT_SECRET")
+	)
+
+	BeforeEach(func() {
+		var err error
+		cli, err = azure.NewClientFromEnvironment(false)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(region).NotTo(BeEmpty())
+		Expect(rg).NotTo(BeEmpty())
+	})
+
+	AfterEach(func() {
+		result, _ := cli.Groups.Delete(ctx, rg)
+		_ = result.WaitForCompletionRef(ctx, cli.Groups.Client())
+	})
+
+	It("should create the vnet and cluster and verify peering", func() {
+		log := tlog.GetTestLogger()
+
+		// create a resource group and a vnet
+		exists, err := cli.Groups.CheckExistence(ctx, rg)
```

Comment:
The resource group is meant to be unique per test run so you can probably remove this check. Reusing groups has been problematic in the past (eg. what if the group is in Deleting state?).

Reply:
Also I think the resource group creation code can be moved in BeforeEach.